### PR TITLE
Add retry counter for waiting on CDI pods

### PR DESCRIPTION
**Pull request by awels**
- If a CDI pod fails it should abort the CI run.

Signed-off-by: Alexander Wels <awels@redhat.com>

### CODE REVIEW

File: hack/build/run-functional-tests.sh [hunk 1]
```diff
@@ -38,8 +38,16 @@ arg_oc="${KUBECTL:+-oc-path=$KUBECTL}"
...
 echo 'Wait until all CDI Pods are ready'
-while [ -n "$(./cluster/kubectl.sh get pods -n $CDI_NAMESPACE -o'custom-columns=status:status.containerStatuses[*].ready' --no-headers | grep false)" ]; do
-    sleep 5
+retry_counter=0
+while [ $retry_counter -lt 30 ] && [ -n "$(./cluster/kubectl.sh get pods -n $CDI_NAMESPACE -o'custom-columns=status:status.containerStatuses[*].ready' --no-headers | grep false)" ]; do
+	retry_counter=$((retry_counter + 1))
+    sleep 10
 done
 
+if [ $retry_counter eq 30 ]; then
```
Comment (jeffvance) [inline hack/build/run-functional-tests.sh]: might be nice if 30 is a readonly defined at the beginning of script so it's easier to find and change.

File: hack/build/run-functional-tests.sh [hunk 2]
```diff
@@ -38,8 +38,16 @@ arg_oc="${KUBECTL:+-oc-path=$KUBECTL}"
 test_args="${test_args} -ginkgo.v ${arg_master} ${arg_namespace} ${arg_kubeconfig} ${arg_kubectl} ${arg_oc}"
 
 echo 'Wait until all CDI Pods are ready'
-while [ -n "$(./cluster/kubectl.sh get pods -n $CDI_NAMESPACE -o'custom-columns=status:status.containerStatuses[*].ready' --no-headers | grep false)" ]; do
-    sleep 5
+retry_counter=0
+while [ $retry_counter -lt 30 ] && [ -n "$(./cluster/kubectl.sh get pods -n $CDI_NAMESPACE -o'custom-columns=status:status.containerStatuses[*].ready' --no-headers | grep false)" ]; do
```
Comment (jeffvance) [inline hack/build/run-functional-tests.sh]: might be nice if 30 is a readonly defined at the beginning of script so it's easier to find and change.

File: hack/build/run-functional-tests.sh [hunk 3]
```diff
@@ -38,8 +38,16 @@ arg_oc="${KUBECTL:+-oc-path=$KUBECTL}"
...
+	retry_counter=$((retry_counter + 1))
+    sleep 10
```
Comment (jeffvance) [inline hack/build/run-functional-tests.sh]: ditto for 10

File: hack/build/run-functional-tests.sh [hunk 4]
```diff
@@ -38,8 +38,16 @@ arg_oc="${KUBECTL:+-oc-path=$KUBECTL}"
 test_args="${test_args} -ginkgo.v ${arg_master} ${arg_namespace} ${arg_kubeconfig} ${arg_kubectl} ${arg_oc}"
 
 echo 'Wait until all CDI Pods are ready'
-while [ -n "$(./cluster/kubectl.sh get pods -n $CDI_NAMESPACE -o'custom-columns=status:status.containerStatuses[*].ready' --no-headers | grep false)" ]; do
-    sleep 5
+retry_counter=0
+while [ $retry_counter -lt 30 ] && [ -n "$(./cluster/kubectl.sh get pods -n $CDI_NAMESPACE -o'custom-columns=status:status.containerStatuses[*].ready' --no-headers | grep false)" ]; do
+	retry_counter=$((retry_counter + 1))
```
Comment (jeffvance) [inline hack/build/run-functional-tests.sh]: this works in most shells: `((retry_coiunter++))`

File: hack/build/run-functional-tests.sh [hunk 5]
```diff
@@ -38,8 +38,16 @@ arg_oc="${KUBECTL:+-oc-path=$KUBECTL}"
...
+    sleep 10
 done
 
+if [ $retry_counter eq 30 ]; then
+	echo "Not all CDI pods became ready"
+	./cluster/kubectl.sh get pods -n $CDI_NAMESPACE
+	exit 1
+git adfi
```
Comment (jeffvance) [inline hack/build/run-functional-tests.sh]: ?

Comment (jeffvance) [reply hack/build/run-functional-tests.sh]: waiting to merge when previous review comments are addressed...

File: hack/build/run-functional-tests.sh (hunk 2 above)
Comment (awels) [reply hack/build/run-functional-tests.sh]: Done

File: hack/build/run-functional-tests.sh (hunk 4 above)
Comment (awels) [reply hack/build/run-functional-tests.sh]: Done

File: hack/build/run-functional-tests.sh (hunk 3 above)
Comment (awels) [reply hack/build/run-functional-tests.sh]: Done

File: hack/build/run-functional-tests.sh (hunk 5 above)
Comment (awels) [reply hack/build/run-functional-tests.sh]: Yeah I am not sure what that is, should be fi....

**Comment by jeffvance** [comment]
lgtm
waiting for ci

**Pull request merged**